pkgs/development/libraries/ncurses/default.nix

```diff
@@ -46,6 +46,9 @@ stdenv.mkDerivation rec {
   buildInputs = lib.optional (mouseSupport && stdenv.isLinux) gpm;
 
   preConfigure = ''
+    # These paths end up in the default lookup chain.
+    export TERMINFO_DIRS=/etc/terminfo
+
     export PKG_CONFIG_LIBDIR="$dev/lib/pkgconfig"
```

Comment:
Because you insisted, I made several tests with different shells and indeed:

```bash
a="a B=b"
export A=$a
echo $B # =>
echo $A # => a B=b
```

I would consider myself quite knowledgeable in bash, but I have no idea why this works so.

Bash documentation has 
```
DEFINITIONS
       [...]
       word   A sequence of characters considered as a single unit by the shell.  Also known as a token.
       name   A word consisting only of alphanumeric characters and underscores, and beginning with an alphabetic  character  or  an  underscore.
              Also referred to as an identifier.
       [...]

BUILTIN COMMANDS
       [...]
       export [-fn] [name[=word]] ...
       export -p
              The supplied names are marked for automatic export to the environment of subsequently executed  commands.   If  the  -f  option  is
              given, the names refer to functions.  If no names are given, or if the -p option is supplied, a list of names of all exported vari‐
              ables is printed.  The -n option causes the export property to be removed from each name.  If a variable name is followed by =word,
              the  value  of the variable is set to word.  export returns an exit status of 0 unless an invalid option is encountered, one of the
              names is not a valid shell variable name, or -f is supplied with a name that is not a function.
```

There is no clue in there that it works as it does. The actual semantics is more like variable affectations optionally preceded by the export keyword.

Bash is a strange language, with a lot of pitfalls. As you can see from both Ralith and me, this corner case is not well known. We assumed that variable expansion occurs here like it does for other commands. The builtin implementation behave differently, but is compatible with our mental model. Why not go the safe side ? Especially when it is not clearly documented.